syntax = "proto3";

package auth.v1;

import "google/protobuf/empty.proto";

service AuthService {
    rpc SignUp(SignUpRequest) returns (SignUpResponse) {}
    rpc SignIn(SignInRequest) returns (SignInResponse) {}
    rpc SignOut(google.protobuf.Empty) returns (google.protobuf.Empty) {}
}

message SignUpRequest {
    string user_name = 1;
    string password = 2;
}

message SignUpResponse {
    string user_id = 1;
}

message SignInRequest {
    string user_name = 1;
    string password = 2;
}

message SignInResponse {
    string user_id = 1;
    string access_token = 2;
}
